libraries/FunctionalInterrupt/examples/Functional/Functional.ino

```diff
@@ -0,0 +1,75 @@
+// Hint: On ESP8266, this no longer depends on FunctionalInterrupt, but is rather a feature of
+// the standard API in Arduino.h
+
+//#include <FunctionalInterrupt.h>
+
+#ifndef IRAM_ATTR
+#define IRAM_ATTR ICACHE_RAM_ATTR
+#endif
+
+#if defined(ESP32)
+#define BUTTON1 16
+#define BUTTON2 17
+#elif defined(ARDUINO_ESP8266_WEMOS_D1MINI)
+#define BUTTON1 D4
+#define BUTTON2 D3
+#else
+#define BUTTON1 2
+#define BUTTON2 0
+#endif
+
+class Button {
+  public:
+    Button(uint8_t reqPin) : PIN(reqPin) {
+      pinMode(PIN, INPUT_PULLUP);
+      //attachInterruptArg(PIN, reinterpret_cast<void(*)(void*)>(&buttonIsr_static), this, FALLING);
+      attachInterrupt(PIN, std::bind(&Button::buttonIsr, this), FALLING);
+    };
+    ~Button() {
+      detachInterrupt(PIN);
+    }
+
+    void IRAM_ATTR buttonIsr() {
+      numberKeyPresses += 1;
+      pressed = true;
+    }
+
+    static void IRAM_ATTR buttonIsr_static(Button* const self) {
+      self->buttonIsr();
+    }
+
+    uint32_t checkPressed() {
+      if (pressed) {
+        Serial.printf("Button on pin %u has been pressed %u times\n", PIN, numberKeyPresses);
+        pressed = false;
+      }
+      return numberKeyPresses;
+    }
+
+  private:
+    const uint8_t PIN;
```

Comment:
I know this is only an example, but let's try to maintain style: please change the naming convention of the members, e.g.: mPin, mNumberKeyPresses, mPressed.

Reply:
decided on m_PIN, m_numberKeyPresses, et cetera - hope that's OK, too.

Reply:
Actually, I was wrong, the correct style we have adopted for this is underscore, e.g.: _PIN, _NumberKeyPresses, etc. Apologies for the confusion, I was reading other code in between.